addons/test_mail/tests/test_mail_composer.py

```diff
@@ -1587,6 +1587,86 @@ def test_mail_composer_delete(self):
         self.assertEqual(len(self._new_msgs), 2, 'Should have created 1 mail.mail per record')
         self.assertFalse(self._new_msgs.exists(), 'Should have deleted mail.message records')
 
+    @users('employee')
+    @mute_logger('odoo.tests', 'odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
+    def test_mail_composer_duplicates(self):
+        """ Ensures emails sent to the same recipient multiple times
+            are only sent when they are not duplicates
+        """
+        self.template.write({
+            'auto_delete': False,
+            'body_html': '<p>Common Body</p>',
+            'email_to': 'test@test.lan',
+            'subject': 'Common Subject',
+            'partner_to': '',
+        })
+        # Guarantee points of variation
+        self.test_records[0].write({'name': 'A'})
+        self.test_records[1].write({'name': 'B'})
+
+        def get_composer():
+            composer_form = Form(self.env['mail.compose.message'].with_context(
+                self._get_web_context(self.test_records, add_web=True,
+                                      default_template_id=self.template.id)
+            ))
+            return composer_form.save()
+
+        composer = get_composer()
+        # no difference
+        with self.mock_mail_gateway(mail_unlink_sent=False), self.mock_mail_app():
+            composer._action_send_mail()
+        self.assertEqual(len(self._mails), 1, 'Should have sent 1 email, and skipped a duplicate.')
+
+        composer = get_composer()
+        # different subject
+        self.template.write({
```

Comment:
Could be great to use subTest in a loop for the various test case to check (Tde loves that). Just to avoid a "too long test", more a "case-based" test.

Reply:
A "unit" test, so to say? ^^ will do

Reply:
They don't have a submarine emoji... sad tde is sad.